### PR TITLE
Esas/stop scenariorun prod 7890

### PR DESCRIPTION
**Pull request by esasova**
Before creating a PR, please check that:
- [ ] Code is clean
- [ ] Tests are still working (cypress tests and `yarn test`)
- [ ] Changes don't cause new errors or warnings in browser console
- [ ] Documentation is up-to-date
- [ ] Breaking changes are clearly identified with [conventional commits](https://kapeli.com/cheat_sheets/Conventional_Commits.docset/Contents/Resources/Documents/index)

### CODE REVIEW

File: src/state/sagas/scenarioRun/StopScenarioRun/StopScenarioRun.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+// generators function
```
Comment (Guismus) [inline src/state/sagas/scenarioRun/StopScenarioRun/StopScenarioRun.js:4]: Wrong line to declare this comment

File: src/state/sagas/scenarioRun/StopScenarioRun/StopScenarioRun.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+          id: action.scenarioRunId,
+        })
+      )
+    );
+  }
+}
+// generators function
+// Here is a watcher that takes EVERY action dispatched named LAUNCH_SCENARIO
+// and binds createScenario saga to it
+function* stopScenarioRunSaga() {
```
Comment (Guismus) [inline src/state/sagas/scenarioRun/StopScenarioRun/StopScenarioRun.js:39]: Bad comment description i think, this action seem stop and not launch

File: src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export const SaveScenarioButton = () => {
+  const { t } = useTranslation();
+  const { isDirty } = useFormState();
+
+  const currentScenarioData = useCurrentScenarioData();
+  const saveScenario = useSaveScenario();
+  const { processFilesToUpload, getParametersToUpdate } = useUpdateParameters();
+  const userAppAndCurrentScenarioPermissions = useUserAppAndCurrentScenarioPermissions();
+
+  const saveScenarioParameters = async (event) => {
```
Comment (ardvi) [inline src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js:24]: usecallback here ;)

File: src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import { ACL_PERMISSIONS } from '../../../../../../services/config/accessControl';
+import { PermissionsGate } from '@cosmotech/ui';
+import { useUserAppAndCurrentScenarioPermissions } from '../../../../../../hooks/SecurityHooks';
+import { useUpdateParameters } from '../../../../../../hooks/ScenarioParametersHooks';
+
+export const SaveScenarioButton = () => {
+  const { t } = useTranslation();
+  const { isDirty } = useFormState();
+
+  const currentScenarioData = useCurrentScenarioData();
```
Comment (ardvi) [inline src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js:19]: useCurrentScenarioId it's better here

File: src/components/ScenarioParameters/components/ScenarioActions/components/DiscardScenarioUpdatesButton/DiscardScenarioUpdatesButton.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  const { isDirty } = useFormState();
+  const scenarioResetValues = useScenarioResetValues();
+  const userAppAndCurrentScenarioPermissions = useUserAppAndCurrentScenarioPermissions();
+  const discardUpdatesConfirmationDialogLabels = {
+    title: t('genericcomponent.dialog.scenario.parameters.title'),
+    body: t('genericcomponent.dialog.scenario.parameters.body'),
+    button1: t('genericcomponent.dialog.scenario.parameters.button.cancel'),
+    button2: t('genericcomponent.dialog.scenario.parameters.button.validate'),
+  };
+  const openDiscardUpdatesConfirmationDialog = (event) => {
```
Comment (ardvi) [inline src/components/ScenarioParameters/components/ScenarioActions/components/DiscardScenarioUpdatesButton/DiscardScenarioUpdatesButton.js:29]: useCallback :)

File: src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+
+export const StopScenarioRunButton = () => {
+  const currentScenarioData = useCurrentScenarioData();
+  const stopScenarioRun = useStopScenarioRun();
+  const { t } = useTranslation();
+  const isCurrentScenarioRunning =
+    currentScenarioData?.state === SCENARIO_RUN_STATE.RUNNING ||
+    currentScenarioData?.state === SCENARIO_RUN_STATE.DATA_INGESTION_IN_PROGRESS;
+  const [isStopRunConfirmationDialogOpen, setIsStopRunConfirmationDialogOpen] = useState(false);
+  const confirmationDialogLabels = {
```
Comment (ardvi) [inline src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js:21]: You can use a useMemo here for keep labels instance

File: src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    title: t('genericcomponent.dialog.scenario.stopRun.title', 'Abort simulation run?'),
+    body: t('genericcomponent.dialog.scenario.stopRun.body', 'Do you want to abort the simulation run'),
+    button1: t('genericcomponent.dialog.scenario.stopRun.button.cancel', 'Cancel'),
+    button2: t('genericcomponent.dialog.scenario.stopRun.button.stop', 'Abort'),
+  };
+  const openStopRunConfirmationDialog = (event) => {
```
Comment (ardvi) [inline src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js:27]: useCallback ;)

File: src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+    event.stopPropagation();
+    setIsStopRunConfirmationDialogOpen(true);
+  };
+  const cancelStopScenarioRun = (event) => {
```
Comment (ardvi) [inline src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js:31]: useCallback ;)

File: src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    event.stopPropagation();
+    setIsStopRunConfirmationDialogOpen(false);
+  };
+  const confirmStopScenarioRun = (event) => {
```
Comment (ardvi) [inline src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js:35]: useCallBack

**Review by ardvi — COMMENTED** [review]
Some coment for use useCallback / useMemo

File: src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  };
+
+  return isDirty ? (
+    <PermissionsGate
+      userPermissions={userAppAndCurrentScenarioPermissions}
+      necessaryPermissions={[ACL_PERMISSIONS.SCENARIO.WRITE]}
+    >
+      <Button data-cy="save-button" size="small" startIcon={<SaveIcon />} onClick={saveScenarioParameters}>
+        {t('commoncomponents.button.scenario.parameters.save', 'SAVE ')}
+      </Button>{' '}
```
Comment (csm-thu) [inline src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js:37]: It seems that a trailing space has been converted to `{' '}` here

File: src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    saveScenario(currentScenarioData?.id, getParametersToUpdate());
+  };
+
+  return isDirty ? (
+    <PermissionsGate
+      userPermissions={userAppAndCurrentScenarioPermissions}
+      necessaryPermissions={[ACL_PERMISSIONS.SCENARIO.WRITE]}
+    >
+      <Button data-cy="save-button" size="small" startIcon={<SaveIcon />} onClick={saveScenarioParameters}>
+        {t('commoncomponents.button.scenario.parameters.save', 'SAVE ')}
```
Comment (csm-thu) [inline src/components/ScenarioParameters/components/ScenarioActions/components/SaveScenarioButton/SaveScenarioButton.js:36]: Extra space remaining at the end of the `SAVE` string

File: src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  const currentScenarioData = useCurrentScenarioData();
+  const stopScenarioRun = useStopScenarioRun();
+  const { t } = useTranslation();
+  const isCurrentScenarioRunning =
+    currentScenarioData?.state === SCENARIO_RUN_STATE.RUNNING ||
+    currentScenarioData?.state === SCENARIO_RUN_STATE.DATA_INGESTION_IN_PROGRESS;
+  const [isStopRunConfirmationDialogOpen, setIsStopRunConfirmationDialogOpen] = useState(false);
+  const confirmationDialogLabels = {
+    title: t('genericcomponent.dialog.scenario.stopRun.title', 'Abort simulation run?'),
+    body: t('genericcomponent.dialog.scenario.stopRun.body', 'Do you want to abort the simulation run'),
```
Comment (csm-thu) [inline src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js:23]: Missing question mark in the default label :slightly_smiling_face:

File: src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+    event.stopPropagation();
+    stopScenarioRun(currentScenarioData?.lastRun?.scenarioRunId, currentScenarioData?.id);
+    setIsStopRunConfirmationDialogOpen(false);
+  };
+  useEffect(() => {
+    if (!isCurrentScenarioRunning) setIsStopRunConfirmationDialogOpen(false);
+  }, [isCurrentScenarioRunning]);
+
+  return isCurrentScenarioRunning ? (
+
```
Comment (csm-thu) [inline src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton/StopScenarioRunButton.js:45]: There are eslint errors here

File: src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton.js [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+import { useCurrentScenarioData } from '../../../../../state/hooks/ScenarioHooks';
+import { SCENARIO_RUN_STATE } from '../../../../../services/config/ApiConstants';
+
+export const StopScenarioRunButton = () => {
+  const currentScenarioData = useCurrentScenarioData();
+  const stopScenarioRun = useStopScenarioRun();
+  const { t } = useTranslation();
+  const isCurrentScenarioRunning =
+    currentScenarioData?.state === SCENARIO_RUN_STATE.RUNNING ||
+    currentScenarioData?.state === SCENARIO_RUN_STATE.DATA_INGESTION_IN_PROGRESS;
```
[the remark this answers is not in the document]
Comment (esasova) [reply src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton.js]: It was in ticket's specs : _scenario runs can be canceled even during DataIngestionInProgress status_ :slightly_smiling_face:

[the remark this answers is not in the document]
Comment (csm-thu) [reply src/components/ScenarioParameters/components/ScenarioActions/components/StopScenarioRunButton.js]: @nborde-CSM if the scenario run is stuck during the _data ingestion_ step, we need to be able to stop it :slightly_smiling_face:

**Review by csm-thu — APPROVED** [review]
LGTM :+1:

**Review by ardvi — APPROVED** [review]
LGTM :+1:

**Pull request merged**